File: Project/ExamResult.cpp
#include "ExamResult.hpp"

#include <stdexcept>


ExamResult::ExamResult(int score) {
	if (score < MIN_SCORE || score > MAX_SCORE) {
		throw std::invalid_argument("Invalid score value");
	}
	m_score = score;
}

std::string ExamResult::ToString() const {
	return std::to_string(m_score);
}

std::string ExamResult::ToStringCompact() const {
	return std::to_string(m_score);
}

int ExamResult::ToPercent() const {
	return ((m_score - MIN_SCORE) * MAX_PERCENT / (MAX_SCORE - MIN_SCORE));
}

bool ExamResult::IsPassed() const {
	return m_score >= PASS_SCORE;
}

int ExamResult::ToScore() const {
	return m_score;
}

File: ProjectCS/ExamResult.cs
using System;

namespace SessionResults
{
    // Результат экзамена
    public class ExamResult : AttestationResult
    {
        public const int MIN_SCORE = 0;
        public const int MAX_SCORE = 100;
        public const int PASS_SCORE = 25;


        // Полученный балл за экзамен
        public int Score { get; }


        public ExamResult(int score) {
            Score = score;
        }


        public override string ToString()
        {
            return Score.ToString();
        }

        public override string ToStringCompact()
        {
            return Score.ToString();
        }

        public override int ToPercent()
        {
            return ((Score - MIN_SCORE) * MAX_PERCENT / (MAX_SCORE - MIN_SCORE));
        }

        public override bool IsPassed()
        {
            return Score >= PASS_SCORE;
        }

        public override int ToScore()
        {
            return Score;
        }
    }
}
